README.rst

```diff
@@ -0,0 +1,227 @@
+  composition of flows results in a flow, it's easy to include one complex
+  workflow as part of another.
+
+Defining a workflow
+-------------------
+
+A simple workflow can be defined from basic Haskell functions::
+
+  -- | A flow which doubles and reverses a list
+  doubleAndReverse :: SimpleFlow [Int] [Int]
+  doubleAndReverse = mapA (step (*2)) >>> step reverse
+
+The `step` function takes a Haskell function of type `a -> b` and lifts it into
+a `SimpleFlow a b`. [1]_ The `mapA` combinator applies a flow over a list, and
+`>>>` composes two flows sequentially.
+
+We can also use Haskell's arrow notation to compose more complex flows::
+
+  -- | Flow which asks for the user's name, then their favourite number.
+  --   The favourite number will be cached, such that if the user comes back,
+  --   they will not be asked again.
+  favouriteNumber = proc () -> do
+    name <- stepIO (\_ -> putStrLn "Hi, what's your name?" >> const getLine) -< ()
+    fnum <- stepIO' (def { cache = defaultCacherWithIdent 172652 })
+                    (\_ -> putStrLn "What's your favourite number?" >> const getLine) -< name
+    returnA -< (name, fnum)
+
+In this flow, we can also see the use of FunFlow's caching. `stepIO` lets us lift
+an `IO` action into a flow. `stepIO'` is a more complex variant that lets us configure
+things such as the caching properties.
+
+Running a workflow
+------------------
+
+To run a workflow, use one of the options from Control.FunFlow.Exec.Simple_ .
+The easiest place to start is with `withSimpleLocalRunner`. This takes the
+absolute path to a directory used to host the content store. It returns a
+function which can be used to run your workflow.
+
+Here's an example of a complete pipeline, along with a runner::
+
+  {-# LANGUAGE Arrows #-}
+  {-# LANGUAGE QuasiQuotes #-}
+  import Control.Arrow
+  import Control.FunFlow
+  import Control.FunFlow.Exec.Simple (withSimpleLocalRunner)
+  import Data.Default
+  import Path
+
+  favouriteNumber :: SimpleFlow () (String, String)
+  favouriteNumber = proc () -> do
+    name <- stepIO (\_ -> putStrLn "Hi, what's your name?" >> getLine) -< ()
+    fnum <- stepIO' (def { cache = defaultCacherWithIdent 172652 })
+                    (\_ -> putStrLn "What's your favourite number?" >> getLine) -< name
+    returnA -< (name, fnum)
+
+  runFavouriteNumber :: IO ()
+  runFavouriteNumber = do
+    res <- withSimpleLocalRunner [absdir|/tmp/funflow|] $ \run ->
+      run favouriteNumber ()
+    case res of
+      Left err -> putStrLn $ "Something went wrong: " ++ show err
+      Right (name, num) -> putStrLn $ "Hi, " ++ name ++ ", your favourite number is " ++ num
+
+Defining external tasks
+-----------------------
+
+External tasks are used to run steps outside of the Haskell process.
+Fundamentally, an external task will resolve to a command-line call to another
+application. This can obviously be done with `stepIO`, but using external tasks
```

Comment:
Suggest replacing "obviously", as it's a value judgement.

"While you could do this with `stepIO`, using external tasks..."